### ISSUE TITLE
Difficulties running on Windows (Release ZIP issues, build issues)

### ISSUE THREAD

**Issue by thepelkus-too** [DBraun/ChucKDesigner#10]
Hi!

Saw this on the TouchDesigner InSession 18.0 from 2022-02-04 and figured I'd give it a shot, but I'm running into two issues on Windows:

1. The directions in the README say to unzip the latest release and copy some dlls. Unfortunately, I'm unable to unzip the zip files for any of the releases using either the built-in zip capabilities in the file explorer in Windows 11, WinZip, or unzip in WSL2. All of them fail with pretty generic messages.
2. I can't figure out how to build from source on Windows. If I run `cmake` from the root, I get an error about not being able to find `chuck.yy.c`. Figuring that I might need to compile a lex file, I ran `flex` on `chuck.lex` and copied the output to `chuck.yy.c`, which got me past that error (albeit with no guarantee that I did the right thing there) and yielded a `cmake_install.cmake` file. I tried running `cmake -P cmake_install.cmake` at that point and got:

```
[  1%] Building CXX object CMakeFiles/ChucKDesignerShared.dir/thirdparty/chuck/src/core/chuck.cpp.o
(include stack omitted here)
fatal error: AvailabilityMacros.h: No such file or directory
  715 | #include <AvailabilityMacros.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
```

I admit that I'm still pretty poorly versed in make, and that I might be able to get past this point if I were better with it. I'll continue to work on getting past this, but both a working zip and some build instructions would be helpful here; happy to assist in any way that I can toward that end!

Thanks for this project and for all that you do for the TD community. 🙏

**Comment by thepelkus-too** [comment]
Quick followup: realizing that last error should be due to it thinking it's building on mac, which it clearly isn't, which further points to my inadequacy with make tools.

**Comment by thepelkus-too** [comment]
Downloaded the source zip in the releases to see if I could have any more luck there, with at least one good result: saw in the github workflows file that **the command to generate the zip files is actually a tar command**! When I ran `tar xvf ChucKDesigner-windows-latest.zip`, I got dlls! Such relief.

That said, I did also try to follow the build instructions that are in there, and my `cmake` command, while it ran smoothly (after copying over the chuck source, since I couldn't init submodules since I wasn't in a git repo), I didn't get the expected sln in the build dir afterwards, so I'm not sure yet what's going on there.

**tl;dr**: the release zips are actually release tars; still not sure how to build on windows.

(And, since this was originally about the zips, really, we can close this here if you like or keep it open to talk about windows build instructions.)

**Comment by DBraun** [comment]
Thanks for the good troubleshooting. I will fix it to not be a tar next time;;;

It sounds like you're almost there with building locally. Definitely do everything the git clone way if possible. I think that bison and the lex stuff just don't need to be run on Windows. If you upload a cmake log/output I can take a look.

**Comment by thepelkus-too** [comment]
You're right about the bison and lex stuff. I nuked my clone of the repo (since I'd tinkered with it and left it in an unknown state), recloned, and just ran the commands from from the github action through powershell. Worked like a charm. Not sure exactly what I messed up the first time around, but I imagine it had something to do with bungling around in wsl.

Thanks for bearing with me. Looking forward to playing with this and Faust in TD!

**Comment by DBraun** [comment]
Great! Please follow up with any issues in either project :)
